src/coreclr/src/vm/interpreter.cpp

```diff
@@ -9095,16 +9095,22 @@ void Interpreter::DoCallWork(bool virtualCall, void* thisArg, CORINFO_RESOLVED_T
         static ConfigDWORD s_InterpreterHWIntrinsicsIsSupportedFalse;
         if (s_InterpreterHWIntrinsicsIsSupportedFalse.val(CLRConfig::INTERNAL_InterpreterHWIntrinsicsIsSupportedFalse) != 0)
         {
-            if (strcmp(methToCall->GetModule()->GetSimpleName(), "System.Private.CoreLib") == 0 &&
-#ifdef _DEBUG // GetDebugClassName() is only available in _DEBUG builds
+            GCX_PREEMP();
+
+            // Hardware intrinsics are recognized by name.
+            const char* namespaceName = NULL;
+            const char* className = NULL;
+            const char* methodName = m_interpCeeInfo.getMethodNameFromMetadata((CORINFO_METHOD_HANDLE)methToCall, &className, &namespaceName, NULL);
+            if (
 #if defined(TARGET_X86) || defined(TARGET_AMD64)
-                strncmp(methToCall->GetClass()->GetDebugClassName(), "System.Runtime.Intrinsics.X86", 29) == 0 &&
+                strcmp(namespaceName, "System.Runtime.Intrinsics.X86") == 0 &&
 #elif defined(TARGET_ARM64)
-                strncmp(methToCall->GetClass()->GetDebugClassName(), "System.Runtime.Intrinsics.Arm", 29) == 0 &&
+                strcmp(namespaceName, "System.Runtime.Intrinsics.Arm") == 0 &&
 #endif // defined(TARGET_X86) || defined(TARGET_AMD64)
-#endif // _DEBUG
-                strcmp(methToCall->GetName(), "get_IsSupported") == 0)
+                strcmp(methodName, "get_IsSupported") == 0
```

Comment:
Is this going to get nested classes such as: `System.Runtime.Intrinsics.Arm.AdvSimd.Arm64.get_IsSupported`?

Reply:
Likewise, given that `DoGetIsSupported()` just returns `false`, is there any reason why we aren't just special casing all of these `get_IsSupported` functions?

Reply:
> Is this going to get nested classes such as: System.Runtime.Intrinsics.Arm.AdvSimd.Arm64.get_IsSupported?

I think these might need to be `strncmp`s to handle all those cases correctly and just check the first 29 chars.

>  is there any reason why we aren't just special casing all of these get_IsSupported functions?

Could you expand what you mean by "special casing" here?  I think this is intended to be a temporary solution until a more production quality approach can be designed.  See [this comment](https://github.com/dotnet/runtime/pull/37825#issuecomment-643630137) for context on what kinds of cases need to be covered by a more robust solution.

Reply:
I was just indicating that they were being handled here to explicitly return false.

Reply:
Gotcha!  Yeah, the intention was to special case _all_ the HW intrinsics' `get_IsSupported` methods to return false until the interpreter has a more robust solution.